File: frontend/src/types/asset.ts
export type AssetStatus = 'FREE' | 'USING' | 'REPAIR' | 'SCRAP'

export interface AssetItem {
  id: number
  assetNo: string
  name: string
  category: string
  price: number
  purchaseDate: string
  status: AssetStatus
  imageUrl?: string
  remark?: string
  currentUserName?: string
  scrapReason?: string
  scrappedAt?: string
  createdAt: string
}

export interface AssetPage {
  page: number
  size: number
  total: number
  records: AssetItem[]
}

export interface AssetQuery {
  page: number
  size: number
  keyword?: string
  category?: string
  status?: AssetStatus
}

export interface AssetSaveReq {
  assetNo: string
  name: string
  category: string
  price: number
  purchaseDate: string
  imageUrl?: string
  remark?: string
}

export interface FileUploadResult {
  fileName: string
  url: string
  size: number
}

File: backend/src/main/java/org/example/eams/entity/Asset.java
package org.example.eams.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;
import org.example.eams.enums.AssetStatus;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@TableName("asset")
public class Asset {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("asset_no")
    private String assetNo;

    private String name;
    private String category;
    private BigDecimal price;

    @TableField("purchase_date")
    private LocalDate purchaseDate;

    private AssetStatus status;

    @TableField("current_user_id")
    private Long currentUserId;

    @TableField("image_url")
    private String imageUrl;

    private String remark;

    @TableField("scrap_reason")
    private String scrapReason;

    @TableField("scrapped_at")
    private LocalDateTime scrappedAt;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;
}
